flink-streaming-java/src/main/java/org/apache/flink/streaming/runtime/tasks/OperatorChain.java

```diff
@@ -409,7 +410,7 @@ public OP getHeadOperator() {
 
 		if (selectors == null || selectors.isEmpty()) {
 			// simple path, no selector necessary
-			if (allOutputs.size() == 1) {
+			if (allOutputs.size() == 1 && !(allOutputs.get(0).f0 instanceof RecordWriterOutput)) {
```

Comment:
This change will force using `BroadcastingOutputCollector` always, which I would expect to have a performance impact. Why not to simply use `CountingOutput` and/or reuse operator level metric in this case?

Reply:
If we use `CountingOutput` directly, we can’t distinguish whether the record is actually sent (just like it was filtered by `OutputTag` in `sideOutput`)